controllers/mongodb_controller.go

```diff
@@ -477,6 +479,12 @@ func (r *MongoDBReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	}
 	r.Log.Info("Mongodb is ready")
 
+	// Be sure to update status before returning if MongoDB is found
```

Comment:
This `defer` statement needs to be earlier in your function; using the `defer` statement here isn't any different than running the function that is being deferred directly, but the goal is to have this function run before any relevant returns occur so that the code is guaranteed to keep the resource up-to-date.

Reply:
@rwhundley , Yes, initially i kept it in the beginning, then i noticed, StatefulSet status was not getting updated due to below error frequently occurring. any idea to solve/mitigate this ?
```
1.6795777186745367e+09	ERROR	controllers.MongoDB.SetService	Attempt to update failed	{"mongodb": "ibm-common-services/ibm-mongodb", "error": "Operation cannot be fulfilled on mongodbs.operator.ibm.com \"ibm-mongodb\": the object has been modified; please apply your changes to the latest version and try again"}
github.com/IBM/ibm-mongodb-operator/controllers.(*MongoDBReconciler).Reconcile.func1
	/workspace/controllers/mongodb_controller.go:121
```

Reply:
> This `defer` statement needs to be earlier in your function; using the `defer` statement here isn't any different than running the function that is being deferred directly, but the goal is to have this function run before any relevant returns occur so that the code is guaranteed to keep the resource up-to-date.

Now i moved the code to beginning